### DISCUSSION TITLE
input demo, how to set icon position

### DISCUSSION THREAD

**Discussion by cy920820** [rexrainbow/phaser3-rex-notes#426] [General]
Can the icon in the input demo be set to the rightmost position within the text box

https://codepen.io/rexrainbow/pen/MWOjJzZ

**Comment by rexrainbow** [comment]
Yes it can, insert `rtl: true` at line 61 and line 73. 
[Reference](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/ui-label/#add-label-object), search `rtl` keyword

**Reply by cy920820** [reply]
It's effective Thank you

**Comment by cy920820** [comment]
How to disable the input box and reopen it after the request is sent? @rexrainbow

**Reply by cy920820** [reply]
```js
    const editor = scene.rexUI.edit(promptsField.getElement('text'), config)
    const editorDom = editor.inputText.node
    this.editorDom.disabled = true
```
I tried this method, but found that it didn't take effect

**Reply by rexrainbow** [reply]
To close editor manually, try `userNameField._edit.close()`. It will be closed automatically when clicking outside of text, or press enter key if `enterClose` is enabled.
[Reference](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/hiddeninputtext/#close-editor).

**Reply by cy920820** [reply]
well，thanks a lot. I will try

**Comment by cy920820** [comment]
I found that the input box will be automatically removed from the DOM tree, which can cause many strange phenomena to occur

**Reply by cy920820** [reply]
For example, when typing on the mobile phone for the first time, the keyboard does not push the input box up, but directly covers the input box

**Reply by cy920820** [reply]
Also, the content of the input box cannot be cleared through JS

**Reply by cy920820** [reply]
Can't input persistence be guaranteed?

**Reply by rexrainbow** [reply]
userNameField, and passwordField have text property, you can change this value after closing editor, including clearing content.
When editing (text-input), text property is set by hiddenEdit behavior.

**Comment by rexrainbow** [comment]
[Here](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/list-inputtext/) is a list of creating a text-input element. First one ([Input text game object](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/inputtext/)) is a pure dom element solution

**Reply by cy920820** [reply]
Very useful, but I am concerned about the final application volume. Can we ultimately support Chunk on-demand packaging

**Reply by cy920820** [reply]
https://codepen.io/rexrainbow/pen/WBxveQ

I have a question, how can I add some icons to the input box, such as clicking on the icon to send the content in the input box?

**Reply by rexrainbow** [reply]
Put input text element into sizer.

**Comment by rexrainbow** [comment]
Might try [DOM game object](https://labs.phaser.io/index.html?dir=game%20objects/dom%20element/&q=) for ui if you are familiar with dom/css.

**Reply by cy920820** [reply]
thanks, Indeed, the dom game object can solve the problem

**Comment by rexrainbow** [comment]
> Very useful, but I am concerned about the final application volume. Can we ultimately support Chunk on-demand packaging

Discussed in post #421 already.

**Reply by cy920820** [reply]
sorry
